migration/identity-migration/src/main/java/io/camunda/migration/identity/TenantMappingRuleMigrationHandler.java

```diff
@@ -0,0 +1,84 @@
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.migration.identity;
+
+import io.camunda.migration.identity.dto.Tenant;
+import io.camunda.migration.identity.dto.TenantMappingRule;
+import io.camunda.migration.identity.midentity.ManagementIdentityClient;
+import io.camunda.migration.identity.midentity.ManagementIdentityTransformer;
+import io.camunda.migration.identity.midentity.MigrationStatusUpdateRequest;
+import io.camunda.migration.identity.service.MappingService;
+import io.camunda.migration.identity.service.TenantService;
+import io.camunda.search.entities.TenantEntity;
+import java.util.List;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TenantMappingRuleMigrationHandler implements MigrationHandler {
+
+  public static final int SIZE = 100;
+  private final ManagementIdentityClient managementIdentityClient;
+  private final ManagementIdentityTransformer managementIdentityTransformer;
+  private final TenantService tenantService;
+  private final MappingService mappingService;
+
+  public TenantMappingRuleMigrationHandler(
+      final ManagementIdentityClient managementIdentityClient,
+      final ManagementIdentityTransformer managementIdentityTransformer,
+      final TenantService tenantService,
+      final MappingService mappingService) {
+    this.managementIdentityClient = managementIdentityClient;
+    this.managementIdentityTransformer = managementIdentityTransformer;
+    this.tenantService = tenantService;
+    this.mappingService = mappingService;
+  }
+
+  @Override
+  public void migrate() {
+    TenantMappingRule lastTenantMappingRule = null;
+    do {
+      final List<TenantMappingRule> tenantMappingRules =
+          managementIdentityClient.fetchTenantMappingRules(lastTenantMappingRule, SIZE);
+      managementIdentityClient.updateMigrationStatus(
+          tenantMappingRules.stream().map(this::createTenantMappingRule).toList());
+      lastTenantMappingRule =
+          tenantMappingRules.isEmpty()
+              ? null
+              : tenantMappingRules.get(tenantMappingRules.size() - 1);
+    } while (lastTenantMappingRule != null);
+  }
+
+  private MigrationStatusUpdateRequest createTenantMappingRule(
+      final TenantMappingRule tenantMappingRule) {
+    try {
+      final var mappingKey =
+          mappingService.findOrCreateMapping(
+              tenantMappingRule.getName(),
+              tenantMappingRule.getClaimName(),
+              tenantMappingRule.getClaimValue());
+      for (final Tenant mappingTenant : tenantMappingRule.getAppliedTenants()) {
+        final var tenant =
+            tenantService.fetchOrCreateTenant(mappingTenant.tenantId(), mappingTenant.name());
```

Comment:
:question: Why is this needed? The tenant migration always runs first so all tenant should exist already. If it's just to fetch the tenant key, I'd try and make that explicit, i.e. split `fetchOrCreateTenant` in two, `fetch` and `create`.

Reply:
yes, but I wanted to reuse it for TenantMigrationHandler when we create the tenant and also with this we are more flexible in ordering the handlers.

Reply:
Since the refresh interval of ES indices is 1 second, I think there's a good chance that we will first create all tenants and then try to create them again here because we can't find them in ES yet. That would lead to CONFLICT errors which we don't handle here so the migration app would just fail.

Reply:
Conflict here is managed as an error status and we don't throw it to the caller, so in next loop for fetching the tenants we have them and can try. By the way do you think it's a good idea to have a pause (sleep) for a few seconds when we first experienced this error? so we have less conflict error.

Reply:
In my new PR, I have separated the `create` and `fetch` methods and I hope it prevent congesting zeebe with a lot of conflict requests.